src/document/invalid-doc-error.js

```diff
@@ -55,6 +51,17 @@ function getDocErrorMessage(doc) {
     throw new Error("doc is valid.");
   }
 
+  const type = doc === null ? null : typeof doc;
+  if (type !== "string" && type !== "object") {
+    return `Unexpected doc '${type}', \nExpected it to be 'string', 'array', or 'object'.`;
```

Comment:
It can't be `string` at this point, and "array" isn't a `typeof` value. 🤓

Reply:
> It can't be string at this poin

It can be if `new InvalidDocError('foo')`

> and "array" isn't a typeof value.

I think it's fine to use in the message.